vote/src/vote_parser.rs

```diff
@@ -43,7 +41,12 @@ pub fn parse_vote_transaction(tx: &Transaction) -> Option<ParsedVote> {
 fn parse_vote_instruction_data(
     vote_instruction_data: &[u8],
 ) -> Option<(VoteTransaction, Option<Hash>)> {
-    match limited_deserialize(vote_instruction_data).ok()? {
+    match limited_deserialize(
+        vote_instruction_data,
+        solana_packet::PACKET_DATA_SIZE as u64,
+    )
```

Comment:
god I hate our `sdk` code lol.

There's a `sdk/program_utils.rs` which has `limited_deserialize`, which is what this used to call.
That fn is just a wrapper around what this is now doing:
```rust
pub fn limited_deserialize<T>(instruction_data: &[u8]) -> Result<T, InstructionError>
where
    T: serde::de::DeserializeOwned,
{
    solana_program::program_utils::limited_deserialize(
        instruction_data,
        crate::packet::PACKET_DATA_SIZE as u64,
    )
}
```

that fn is itself just a wrapper around bincode::deserialize_from with specific options set...or re-created each time.

Change here is simply because that function and the `program_utils module` are still part of `sdk` rather than their own crate.